### DISCUSSION TITLE
Passing styles to Controller (for React Native)

### DISCUSSION THREAD

**Discussion by pke** [react-hook-form/react-hook-form#8393] [Q&A] [answered]
I wonder how it would be possible to pass styles to a Controller (which passes them on to the rendered component)?

Lets say I want to have the form embedded into my layout component, which adds a "marginBottom" to each child component.

```tsx
<Stack>
  <Controller render=({style}) => <Input style={style} />
</Stack>
```

But currently the controller can not receive style as a prop, nor can it forward it.

**Comment by Moshyfawn** [comment, marked as answer]
`Controller` isn't concerned about styes or whatever else props you might have. The only props you can get via the render callback are the field handlers (+ value), the field state _(like `error`)_ and the whole form state _(i.e. `isSubmitSuccessful`)_.

RHF is a headless _(if you will)_ library that handles only the form management.

You can pass your `Input`'s styles directly to your `Input` component

**Reply by bluebill1049** [reply]
Yea pretty much Controller, useController is headless it only collects and updates the value.

**Reply by pke** [reply]
Thanks. So I should use `useController` instead is what you are saying?

btw: the docs page is not working in Chrome (with umatrix). It shows the pages content briefly and then shows a blank page. Scripts are allowed and caches cleared.
